common/src/main/java/io/netty/util/concurrent/DefaultPromise.java

```diff
@@ -420,18 +432,88 @@ private void notifyListeners() {
         }
         EventExecutor executor = executor();
         if (executor.inEventLoop()) {
```

Comment:
Should we really care much about the `ImmediateExecutor` case? Within Netty, `ImmediateExecutor` is only used in a minor way in `SslHandler`, and I have difficulty thinking it is used frequently by user code for Promises. It seems if `ImmediateExecutor` causes stack depth problems (which is can on its own), then maybe it should be fixed in its own way and not worry about it here.